main.go

```diff
@@ -89,11 +79,18 @@ func main() {
 		MacaroonHex: os.Getenv("MACAROON_HEX"),
 	})
 	if err != nil {
-		log.Fatalf("Failed to create LND client: %v", err)
+		fmt.Printf("Failed to create LND client: %v", err)
+	}
+	lnurlClient := &lnurl.LNURLWrapper{
+		Address: os.Getenv("LNURL_ADDRESS"),
+	}
+
+	svc := &service.Service{
+		LndClient:   lndClient,
+		LnurlClient: lnurlClient,
 	}
-	svc.lndClient = lndClient
 
-	router.GET("/protected", svc.getProtectedResource)
+	router.GET("/protected", getProtectedResource(svc))
```

Comment:
I get that you want to return a `gin.HandlerFunc` to move to middleware later, but why did you move the service back to the arguments, instead of defining the function on it?